delira/utils/config.py

```diff
@@ -1,13 +1,509 @@
+import copy
+from delira._version import get_versions
+from delira.utils.time import now
 from nested_lookup import nested_lookup
-from trixi.util import Config
+import warnings
+from .codecs import Encoder, Decoder
+
+import yaml
+import argparse
+
+
+def non_string_warning(func):
+    def warning_wrapper(config, key, *args, **kwargs):
+        """
+        Emit warning if non string keys are used
+
+        Parameters
+        ----------
+        config: :class:`Config`
+            decorated function receive `self` as first argument
+        key : immutable type
+            key which is checked
+
+        Returns
+        -------
+        callable
+            original function with arguments
+        """
+        if not isinstance(key, str):
+            warnings.warn("The key {} is not a string, but a {}. "
+                          "This may lead to unwanted behavior!".format(
+                              key, type(key)), RuntimeWarning)
+
+        return func(config, key, *args, **kwargs)
+    return warning_wrapper
+
+
+class Config(dict):
+    """
+    Baseclass to create a config which hold arbitrary data
+    """
+
+    def __init__(self, dict_like=None, **kwargs):
+        """
+
+        Parameters
+        ----------
+        dict_like : dict, optional
+            dict like object to initialize config, by default None
+        kwargs:
+            additional arguments added to the config
+
+        Warnings
+        --------
+        If the short form of the nested settings is used integers and str
+        can not be differentiated. Therefore, integers should be avoided.
```

Comment:
don't specify this to ints, but more general to other (non-string) types as floats etc. are a problem too